packages/vue/src/components/user-name-alias.vue

```diff
@@ -146,12 +150,28 @@ if (userNameAlias) {
   name = 'username';
 }
 label = translate<string>(label);
-onMounted(() => {
-  if (type === 'tel') {
-    send({
-      type: 'CHANGE',
-      data: { name: 'country_code', value: defaultDialCode },
-    });
-  }
+
+const fullPhoneNumber = ref({
+  country_code: defaultDialCode,
+  [name]: '',
 });
+
+/**
+ * When the field being rendered is for a phone number, this handler is used to prevent change event propagation in
+ * order to manually update the state machine with a country code + phone number.
+ */
+const handlePhoneNumberChange = (event: Event) => {
+  event.stopPropagation();
+
+  const target = <HTMLInputElement | HTMLSelectElement>event.target;
+
+  fullPhoneNumber.value[target.name] = target.value;
+
+  const { country_code, [name]: phoneNumber = '' } = fullPhoneNumber.value;
+
+  updateForm({
+    name,
+    value: country_code + phoneNumber,
+  });
+};
```

Comment:
This looks good, but when I was testing the `signUpAttributes` with `phone_number` it was causing cognito to return back `Invalid phone number format.`.

It doesn't send over the area code in that instance.

![image](https://user-images.githubusercontent.com/65630/150016237-9b8ce9ea-8f2d-49a3-b793-02f454bcffdc.png)

Reply:
It works fine if the phone_number is the primary login mechanism though.

Reply:
Ah this makes sense since we are using the `PhoneNumberField` primitive when phone number is a sign up attribute and not the `UserNameAlias` component. I just realized our e2e tests don't actually test signing up with attributes. I'll look into addressing this problem when phone is an attribute.